components/sup/src/manager.rs

```diff
@@ -1595,6 +1596,40 @@ impl Manager {
                     // Execute the future synchronously
                     self.add_service_rsw_mlw_rhw_msr(spec).await;
                 }
+                ServiceOperation::Update(spec, ops) => {
+                    trace!("ServiceOperation::Update! {:?}", spec);
+                    let mut services = self.state.services.lock_msw();
+                    // Relies on spec.ident not having changed, which
+                    // ServiceSpec#reconcile must guarantee.
+                    let ident = spec.ident.clone();
+
+                    // TODO (CM): While it is technically OK to remove
+                    // the service, modify it, and then re-insert it,
+                    // this is conceptually a modify-in-place
+                    // operation. Ideally, the code would reflect
+                    // this.
+                    if let Some(mut s) = services.remove(&ident) {
+                        s.set_spec(spec);
+
+                        for op in ops {
+                            match op {
+                                RefreshOperation::RestartUpdater => {
+                                    self.service_updater.lock().register(&s);
+                                }
+                            }
+                        }
```

Comment:
Not for this PR. But I could see extracting this into an `update` function. Then on service startup, we would pass in a list of all the various updates. This would prevent us from having duplicated logic between service startup and update.

Reply:
Yup, I think that's the natural next step.